### ISSUE TITLE
Registry S3 storage driver should have ability to use 3rd party S3 compatible cloud

### ISSUE THREAD

**Issue by mastier** [docker/distribution#1315]
There should be ability to add 3rd party S3 compatible private cloud like Riak CS2. Please give me some guidance.
Shall I create new similar driver or modify s3 driver ?

**Comment by stevvooe** [comment]
This is a duplicate of #767 and covered by a PR in https://github.com/docker/distribution/pull/808. Please read the issue tracker before filing such an issue.

It is held up on appropriate testing, so any help there could help push it forward.

**Comment by mastier** [comment]
I have read, but it seems I made a wrong query. Thanks and sorry for duplication!

**Comment by stevvooe** [comment]
@mastier No worries. GitHub has poor search functionality.

Testing with Riak would be great! We want users to have confidence in the storage drivers and any extra testing can help with that.
